internals/src/error/parse_error.rs

```diff
@@ -17,8 +17,9 @@
 /// * `derive` - list of derives to add
 #[macro_export]
 macro_rules! parse_error_type {
-    ($vis:vis $name:ident, $source:ty, $subject:expr $(, $derive:path)* $(,)?) => {
-        #[derive(Debug $(, $derive)*)]
+    ($doc: literal, $vis:vis $name:ident ( $source:ty, $subject:expr ) ) => {
```

Comment:
Concept ACK improving this but it can be better. We can support arbitrary attributes and we should have the `struct` keyword in the macro. The body of the struct is a bit confusing pretending it's tuple struct but I'm not sure how to improve it.

Reply:
PR now does not use this macro so I dropped the changes alltogether.